src/components/Filters/SchemaSieve.js

```diff
@@ -53,11 +53,23 @@ class SchemaSieve {
   }
 
   filterArray (collection, input) {
-    return collection.filter(item => this.baseTest(item, input))
+    if (_.isArray(input)) {
+      return collection.filter(item =>
+        _.every(input, i => this.baseTest(item, i))
+      )
+    } else {
+      return collection.filter(item => this.baseTest(item, input))
+    }
   }
 
   filterObject (collection, input) {
-    return _.pickBy(collection, value => this.baseTest(value, input))
+    if (_.isArray(input)) {
+      return _.pickBy(collection, value =>
+        _.every(input, i => this.baseTest(value, i))
+      )
+    } else {
+      return _.pickBy(collection, value => this.baseTest(value, input))
+    }
```

Comment:
You could more simply refactor the common structure here (if array, `_.every(x => test(x))`, otherwise `test(x)`)  into `baseTest`, or into a new function which wraps `baseTest` (`test` and `testSingleInput`?).

That'll be much simpler, and means this code wouldn't have to change, and you wouldn't end up duplicating the same logic and structure you've used here.

Reply:
Very good point, looking at the code I think it would be best to wrap `baseTest` in a function `test` that will perform the `isArray` check.